lib/ansible/plugins/connection/local.py

```diff
@@ -47,7 +47,11 @@ def __init__(self, *args, **kwargs):
 
         super(Connection, self).__init__(*args, **kwargs)
         self.cwd = None
-        self.default_user = getpass.getuser()
+        try:
+            self.default_user = getpass.getuser()
```

Comment:
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, add tests for this change.

Reply:
we use getpass.getuser() in many places, we might want to make this a generic function in module_utils and replace the current  instances of it, instead of fixing just this one